runtime/src/bank/partitioned_epoch_rewards/mod.rs

```diff
@@ -44,18 +45,44 @@ type PartitionedStakeRewards = Vec<PartitionedStakeReward>;
 pub(crate) struct StartBlockHeightAndRewards {
     /// the block height of the slot at which rewards distribution began
     pub(crate) distribution_starting_block_height: u64,
-    /// calculated epoch rewards pending distribution, outer Vec is by partition (one partition per block)
+    /// calculated epoch rewards before partitioning
+    pub(crate) all_stake_rewards: Arc<Vec<PartitionedStakeReward>>,
+}
+
+#[derive(Debug, Clone, PartialEq)]
+pub(crate) struct StartBlockHeightAndPartitionedRewards {
```

Comment:
We could save on allocation overhead by having this struct store `all_stake_rewards` and then a field for `partitions: Vec<Vec<usize>>` where `usize` is an index into `all_stake_rewards`. What do you think? If you want to explore this, I think it should be a separate PR though.

Reply:
Yes, this is an alternative idea. We had thought about this before. The downside of this approach is that we can't store the accounts in one batch. in account's db. Instead, we need to store the account individually. And storing the account individually is more expensive than storing all the accounts in one batch. And that offsets the benefit of avoid the allocations.

Reply:
We can still store the accounts in one batch with the approach I suggested. Why do you say they need to be stored individually?

Looks like in `fn store_stake_accounts_in_partition` we can look up `&PartionedStakeReward` from `all_stake_rewards` and then pass to `Self::build_updated_stake_reward` before calling `store_accounts(..)`

Reply:
Ok. Now I see it. I will make this change in the new PR too.